packages/router/src/util.ts

```diff
@@ -96,11 +97,15 @@ type SupportedRouterParamTypes = keyof typeof coreParamTypes
  *
  *  matchPath('/post/{id:Int}', '/post/7')
  *  => { match: true, params: { id: 7 }}
+ *
+ *  matchPath('/post/1', '/post/', matchChildRoutes: true)
+ *  => { match: true, params: { }}
  */
 export const matchPath = (
   route: string,
   pathname: string,
-  paramTypes?: Record<string, ParamType>
+  paramTypes?: Record<string, ParamType>,
+  matchChildRoutes = false
```

Comment:
Now that we have so many arguments to this function, could you please switch to an options object instead? Either for all of them, or for just the last two.

Reply:
Wouldn't that change the api? Is it safe to assume, that this is only used internally?

Reply:
That's a good shout! The "public version" is supposed to be the `useMatch` hook. But we are also exposing this function, so while it's ment to be internal only, it really isn't. 

Let's leave it as it is for now and I'll ask the rest of the team. We are about to release RW v4 fairly soon. So if we are to introduce a breaking change that would be a good time to do it :)

Reply:
I switched to an options object, but I also kept the old way of calling the function, but with a deprecation notice. 

The change that this PR introduces is something I need in one of my own projects now, so keen to get it merged. And always easier to merge without breaking changes :D